### PR TITLE
Appointments top section visible when walkins unspecified and not remove_text

### PR DESCRIPTION
**Pull request by eselkin**
This is merging to the Service Location Integration Branch

## Summary

- Makes conditional a liquid filter since conditional is too complex for inline liquid
- Sitewide Facilities

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17516

## Testing done

- Tested on Los Angeles VA Clinic Service Locations from Service Location Integration tugboat

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Desktop | Screenshot 2024-04-08 at 3 14 50 PM       |    Screenshot 2024-04-08 at 3 14 33 PM   |

## What areas of the site does it impact?

Service Location Paragraphs in Accordions in VAMCs

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

Check that logic is appropriate locally (or release on tugboat of integration branch) for 
http://localhost:3002/greater-los-angeles-health-care/locations/los-angeles-va-clinic/#suicide-prevention

### CODE REVIEW

**Comment by davidmpickett** [comment]
Confirmed on Tugboat

https://web-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/greater-los-angeles-health-care/locations/santa-maria-va-clinic/#optometry

Screenshot 2024-04-10 at 5 57 31 PM

File: src/site/filters/liquid.js [hunk 1]
```diff
@@ -1685,6 +1685,28 @@ module.exports = function registerFilters() {
...
+      'yes_appointment_only',
+      'yes_with_or_without_appointment',
+    ];
+    return (
+      yesConditions.includes(virtualSupport) ||
+      yesConditions.includes(officeVisits) ||
+      (!officeVisits &&
+        introTextType === 'customize_text' &&
+        introTextCustom) ||
+      (!officeVisits && introTextType === 'default_text')
```
Comment (davidmpickett) [inline src/site/filters/liquid.js:1706]: @eselkin A bit out of my depth here, but trying to support you while @maxx1128 is out. 

I don't fully understand line 1706. I think I follow the logic up to that point. 

1700 = We show Appointment header and text IF
1701 = Virtual Support is either yes_appointment_only yes_with_or_without_appointment
OR
1702 = Office Visits is either yes_appointment_only yes_with_or_without_appointment
OR
1703/4/5 = Office Visits is Null, but Appointment Intro text is set to Custom and they've actually written something in the custom text box
OR
1706 = Office Visits is Null, and Appointment Intro text is set to Default.

The fourth OR here seems like it should lead to [Cardiology at Perry](https://web-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/dublin-health-care/locations/perry-va-clinic/#cardiology) displaying Appointment header and Default Text, but it isn't currently?

Comment (davidmpickett) [reply src/site/filters/liquid.js:1706]: Update: I think this might need to be 'use_default_text' instead of 'default_text'

**Comment by davidmpickett** [comment]
Full review of new sub logic

## Office Visits = Null && Custom Appointment Text ✅ Correctly showing Appointment header & Text
https://web-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/greater-los-angeles-health-care/locations/santa-maria-va-clinic/#optometry

Screenshot 2024-04-10 at 5 57 31 PM


## Office Visits = Null && Default Appointment Text ❌ Incorrectly NOT showing Appointment Header & text

https://web-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/dublin-health-care/locations/perry-va-clinic/#cardiology

Screenshot 2024-04-10 at 6 36 04 PM


## Office Visits = Null && Remove Appointment Text ✅ Correctly NOT showing Appointment header

https://web-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/san-francisco-health-care/locations/clearlake-va-clinic/#advice-nurse

Screenshot 2024-04-10 at 6 33 50 PM

File: src/site/filters/liquid.js [hunk 2]
```diff
@@ -1685,6 +1685,28 @@ module.exports = function registerFilters() {
...
+      'yes_appointment_only',
+      'yes_with_or_without_appointment',
+    ];
+    return (
+      yesConditions.includes(virtualSupport) ||
+      yesConditions.includes(officeVisits) ||
+      (!officeVisits &&
+        introTextType === 'customize_text' &&
+        introTextCustom) ||
+      (!officeVisits && introTextType === 'use_default_text')
```
Comment (maxx1128) [inline src/site/filters/liquid.js:1706]: **Suggestion (non-blocking):** I would move these last two conditionals to variables. That would simplify the return statement and also make their intent more clear to other programmers coming back to this code later on.

```javascript
// Just example variable names
const noVisitsAndCustomIntro = !officeVisits && introTextType === 'customize_text' && introTextCustom
const noVisitsAndDefaultIntro = !officeVisits && introTextType === 'use_default_text'

return (
    yesConditions.includes(virtualSupport) ||
    yesConditions.includes(officeVisits) ||
    noVisitsAndCustomIntro ||
    noVisitsAndDefaultIntro
)
```

**Review by maxx1128 — APPROVED** [review]
I have one suggestion for refactoring the code, but it's non-blocking, so I'm fine with approving it regardless.

File: src/site/filters/liquid.js (hunk 2 above)
Comment (eselkin) [reply src/site/filters/liquid.js]: Moved to variables.

**Comment by davidmpickett** [comment]
Of the three scenarios above, the one that was previous broken is now fixed
https://web-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/dublin-health-care/locations/perry-va-clinic/#cardiology

Screenshot 2024-04-18 at 5 18 04 PM

**Pull request merged**